### ISSUE TITLE
Formatting if expressions not according to style-guide

### ISSUE THREAD

**Issue by michalkovy** [fsprojects/fantomas#446]
When formatting F# code
```FSharp
if cond then e1 else e2
```
where e1 is multi-line and e2 is short then Fantomas chooses to format it
```FSharp
if cond then 
    e1 
else e2
```
but https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-if-expressions seems to suggest that it should be formatted
```FSharp
if cond then 
    e1 
else 
    e2
```

Similarly if e2 is multi-line and e1 isn't then Fantomas creates
```FSharp
if cond then e1
else
    e2
```

### Repro code

Try to format these using [the online fantomas tool](http://ratatosk.dynu.net/fantomas/):
```FSharp
let private getConfigurationFromFile fileName =
    if File.Exists fileName then
        let config = ConfigFile()
        config.Load(fileName)
        config
    else
        failwithf "File %s was not found" fileName
```

```FSharp
let private getConfigurationFromFile fileName =
    if not File.Exists fileName then 
        failwithf "File %s was not found" fileName
    else
        let config = ConfigFile()
        config.Load(fileName)
        config
```

**Comment by jindraivanek** [comment]
@michalkovy In case you are not aware, you can use online fantomas url to share samples: [1](https://jindraivanek.gitlab.io/fantomas-ui/#?code=DYUwLgBADgTglgNwIZhBA5uAwgewHYBmc6ArjCnPgGIw4C2VcoERoAcknWgLwCwAUBCEQ4BCI1AA6AKIAPOAGcwClkxAcuEMAAsQeAcMMRQkAMb4i6CNwi5CxCSAAUASgNGh5++kkAZHEgAJk6s6pwgboIeEF6W7kIgwAog8UYESEwA7nA6YgBEjhAApCqZSCp4OJAEOCR4gXmq7OFAA), [2](https://jindraivanek.gitlab.io/fantomas-ui/#?code=DYUwLgBADgTglgNwIZhBA5uAwgewHYBmc6ArjCnPgGIw4C2VcoERoAcknWgLwCwAUBCEQ4BCHhyRGoAHQBRAB5wAzmGUsmIDlwhgAFiDwQBw0yyRMA7nH1iARNLQBSdZaTqJkAjhJ4AJnYa7JwgJsIgwMqhgmZCoJAAxvhE6BDcELiExI4AFACUYbFJWegyADI4SH45rFohBTFFycRAA)

Can confirm this behavior and that it is not aligned with style guide.

**Comment by nojaf** [comment]
@jindraivanek, I have a fix for this but I'm unsure if `multiline` check is ok here.
Perhaps `futureNewLine` would be a better fit. Thoughts?

**Comment by jindraivanek** [comment]
@nojaf Yes `futureNlnCheck` would be better, for example `multiline` don't detect new line after `then` in [this](https://jindraivanek.gitlab.io/fantomas-ui/#?fantomas=preview-trivia&code=DYUwLgBADgTglgNwIZhBA5uAwgewHYBmc6ArjCnPgGIw4C2VcoERoAcknWgLwCwAUBCEQ4BCHhyRGoAHQBRAB5wAzmGUsmIDlwhgAFiDwQBw0yyRMA7nH1iARNLQBSdZaTqJkAjhJ4AJnYa7JwgJsIgwMqhgmZCAMb4ROhAA&config=N4IgkgdgJgphAuBlADgQwMYwHIFcC2IAXACwA0IACqgOYwDqAllPABZEBsADORQE4wBnGLwBuMAKLQA8gDMAMgwgwiM1ABsh5RDDwN0AezX6IAQXiSoshUpXrNIFBhgAhGDP38Tvavjjwi8Lw4MFpomK7u-ADChsa2GiEOYTAmMvDCMXh4qAFBiY6Yqem82roGRhC5weSQsAhSEAAqvACejKzx9gBKMB6wvFLIcAAiMOhqqLyo8AxxhKoJoU5e+jjQo2oMusVViQDSMDDIWDAA7ptKRcKd+YF68ACy+rA3AL5AA).

**Comment by jindraivanek** [comment]
I found out that it is not that easy (replace `multiline` by `futureNlnCheck`), because `futureNlnCheck` try to write following expr at current position and then examine if there was some newline. We can't use it for upcoming expr (`else` branch when we are at `if` branch). Something like 2-phase processing of whole `if` block is needed to do it properly.

I think we should close this as good-enough for now and make new backlog issue for the specific cases where `multiline` is not enough.

**Comment by nojaf** [comment]
From time to time I like the good enough approache 😉 it would indeed require a multi phase strategy.
